modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/file/FilePageStoreManager.java

```diff
@@ -1103,6 +1110,23 @@ private IgniteCheckedException shutdown(CacheStoreHolder holder, boolean cleanFi
         return aggr;
     }
 
+    private void cleanParentDirectories(CacheStoreHolder holder) {
+        Arrays.stream(holder.partStores).map(store -> ((FilePageStoreV2)store).getPath().getParent()).distinct().forEach(new Consumer<Path>() {
+            @Override public void accept(Path dir) {
+                if (Files.isDirectory(dir, LinkOption.NOFOLLOW_LINKS)) {
```

Comment:
I believe you can reuse `org.apache.ignite.internal.util.IgniteUtils#delete(java.nio.file.Path)`, it is recursive